embed.fnc

```diff
@@ -3661,6 +3666,33 @@ XEop	|char  *|dup_warnings	|NULLOK char *warnings
 
 Amd	|const char * const|phase_name					\
 				|enum perl_phase
+
```

Comment:
I am confused why none of the new symbols are guarded properly as requested in embed.fnc's doc. Functions that are not public should not be added with no restrictions, otherwise they can be used directly by code that shouldn't on *nix but wont work on Win32 and more restrictive architectures. 

Id expect at least some of the non 'A' functions to be guarded with something like:

```
#if defined(PERL_IN_CLASS_C)
#endif
```
Or possibly with other file name defines listed.  Note I previously requested you add such a define to be compatible with our other source files, this is why it is needed.

Reply:
> otherwise they can be used directly by code that shouldn't on *nix

This is no longer true for ELF systems where the compiler supports the visibility attribute.

Hiding the names is still useful, since it avoids namespace pollution by the macros.

Reply:
Can this conversation be marked resolved now?

Reply:
Nope because I haven't done anything requested in it yet.

Reply:
@demerphq 
> I am confused why none of the new symbols are guarded properly as requested in embed.fnc's doc.

I finally got around to fixing this. It's now ifdef'ed on any of class.c, perly.y, toke.c or pad.c (because some of the field adjustments happen from there).